tests/iana_registries/arp_parameters.rflx

```diff
@@ -0,0 +1,271 @@
+-- AUTOMATICALLY GENERATED BY /home/rs/Dev/RecordFlux-specifications/./tools/iana_to_rflx.py. DO NOT EDIT.
+-- Generation date: 2021-06-17
+-- Address Resolution Protocol (ARP) Parameters
+-- Registry last updated on 2016-07-20
+
+package Arp_Parameters is
+
+   type Operation_Codes is
+      (
+       -- Ref: rfc826, Ref: rfc5227
+       --
+       Request => 1,
+
+       -- Ref: rfc826, Ref: rfc5227
+       --
+       Reply => 2,
+
+       -- Ref: rfc903
+       --
+       Request_Reverse => 3,
+
+       -- Ref: rfc903
+       --
+       Reply_Reverse => 4,
+
+       -- Ref: rfc1931
+       --
+       Drarp_Request => 5,
+
+       -- Ref: rfc1931
+       --
+       Drarp_Reply => 6,
+
+       -- Ref: rfc1931
+       --
+       Drarp_Error => 7,
+
+       -- Ref: rfc2390
+       --
+       Inarp_Request_8 => 8,
+
+       -- Ref: rfc2390
+       --
+       Inarp_Reply_9 => 9,
+
+       -- Ref: rfc1577
+       --
+       Arp_Nak => 10,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Request => 11,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Multi => 12,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Mserv => 13,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Join => 14,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Leave => 15,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Nak => 16,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Unserv => 17,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Sjoin => 18,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Sleave => 19,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Grouplist_Request => 20,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Grouplist_Reply => 21,
+
+       -- Ref: Grenville_Armitage
+       --
+       Mars_Redirect_Map => 22,
+
+       -- Ref: Mitsuru_Maruyama, Ref: rfc2176
+       --
+       Mapos_Unarp => 23,
+
+       -- Ref: rfc5494
+       --
+       Op_Exp1 => 24,
+
+       -- Ref: rfc5494
+       --
+       Op_Exp2 => 25)
+   with Size => 5, Always_Valid;
```

Comment:
The aspect does make sense because 26 to 65534 are Unassigned or Reserved. What was wrong is the type size.
Fixed in fee4364

Reply:
What do you think `Always_Valid` is doing? Having unassigned or reserved values is no reason for having this aspect.

Reply:
> Having unassigned or reserved values is no reason for having this aspect.

Quite the opposite: With `Always_Valid` all values - including the reserved ones - are accepted by the parser. This is undesired - the parser should reject reserved values a invalid.

`Always_Valid` can be useful if you have a protocol field with a few named protocols, but you also want to accept unknown / future protocol IDs (in which case e.g. you only use the raw payload instead of refinement).

Reply:
So the problem is that I generated `arp_parameters.rflx` with the `-a` cli switch set? 
If that argument is not given, `Always_Valid` will never occur on any type in this file. 
From my understanding, it does make sense for the tool to append `Always_Valid` to the types in case the `-a` switch is explicitly set by the user, because in this case, the user wants Reserved and Unassigned values to be accepted.

Reply:
> So the problem is that I generated `arp_parameters.rflx` with the `-a` cli switch set?

Yes, the ARP standard does not specify that unknown values have to be accepted, thus `Always_Valid` must not be used here.

> If that argument is not given, `Always_Valid` will never occur on any type in this file.
> From my understanding, it does make sense for the tool to append `Always_Valid` to the types in case the `-a` switch is explicitly set by the user, because in this case, the user wants reserved and unassigned values to be accepted.

That is correct.

Reply:
Regenerated the specifications in 5812afdd